pyswarms/single/general_optimizer.py

```diff
@@ -256,11 +260,15 @@ def optimize(self, objective_func, iters, n_processes=None, verbose=False, **kwa
             self._populate_history(hist)
             # Verify stop criteria based on the relative acceptable cost ftol
             relative_measure = self.ftol * (1 + np.abs(best_cost_yet_found))
-            delta = np.abs(self.swarm.best_cost - best_cost_yet_found) < relative_measure
+            delta = (
+                np.abs(self.swarm.best_cost - best_cost_yet_found)
+                < relative_measure
+            )
             if i < self.ftol_iter:
-                ftol_history[i] = delta
+                ftol_history.append(delta)
             else:
-                ftol_history = ftol_history[1:] + [delta]
+                ftol_history.popleft()
```

Comment:
We don't know the maximum size of the queue beforehand, right? If we do, maybe we can use the `maxlen` parameter instead of popping it left manually. But if we don't know the queue size, then I think this looks good to me

Ref: https://stackoverflow.com/questions/5944708/python-forcing-a-list-to-a-fixed-size

Reply:
I think that should be possible. The `maxlen = ftol_iter`.
So, I'll set that when initializing the `deque()` object and we can get rid of `popleft()`.

Thanks @ljvmiranda921 good find! I completely overlooked this functionality.